runtime/vm/ValueTypeHelpers.hpp

```diff
@@ -456,6 +456,8 @@ class VM_ValueTypeHelpers {
 
 	/**
 	 * Performs an aaload operation on an object. Handles flattened and non-flattened cases.
+	 * This function could call into J9AllocateObject() which might trigger GC (in the slow path), so receiverObject might be moved by GC.
+	 * If the caller caches receiverObject and uses it after calling this fucntion, it needs to re-read receiverObject.
 	 *
 	 * Assumes recieverObject is not null.
```

Comment:
below `If fast == false, special stack frame must be built and receiverObject must be pushed onto it.` that should change as the caller is no longer responsible for putting the receiverObject in the specialFrame

Reply:
Removed that comment.